Firmware/tmc2130.cpp

```diff
@@ -272,17 +272,15 @@ void tmc2130_home_enter(uint8_t axes_mask)
 		tmc2130_wait_standstill_xy(1000);
 	for (uint8_t axis = X_AXIS; axis <= Z_AXIS; axis++) //X Y and Z axes
 	{
-		uint8_t mask = (X_AXIS_MASK << axis);
-		if (axes_mask & mask)
+		if (const uint8_t mask = axes_mask & _BV(axis); mask)
 		{
 			tmc2130_sg_homing_axes_mask |= mask;
 			//Configuration to spreadCycle
 			tmc2130_wr(axis, TMC2130_REG_GCONF, TMC2130_GCONF_NORMAL);
 			tmc2130_wr(axis, TMC2130_REG_COOLCONF, (((uint32_t)tmc2130_sg_thr_home[axis]) << 16));
 			tmc2130_wr(axis, TMC2130_REG_TCOOLTHRS, __tcoolthrs(axis));
 			tmc2130_setup_chopper(axis, tmc2130_mres[axis], tmc2130_current_h[axis], tmc2130_current_r_home[axis]);
-			if (mask & (X_AXIS_MASK | Y_AXIS_MASK | Z_AXIS_MASK))
```

Comment:
**Note:** I think this if statement is always true since the mask will always include one of the axis bits.